### PR TITLE
Fix ios device bad display effect

### PR DESCRIPTION
**Pull request by timlyeee**
* https://github.com/cocos-creator/3d-tasks/issues/11159

### CODE REVIEW

**Comment by timlyeee** [comment]
测试过相关的issue https://github.com/cocos-creator/3d-tasks/issues/5814 没有问题。

File: templates/ios/ViewController.mm [hunk 1]
```diff
@@ -51,31 +48,21 @@ - (BOOL)prefersHomeIndicatorAutoHidden {
...
-    }
+    cc::EventDispatcher::dispatchOrientationChangeEvent((int) orientation);
+    
+    float    pixelRatio = cc::Device::getDevicePixelRatio();
+    cc::EventDispatcher::dispatchResizeEvent(size.width * pixelRatio
+                                             , size.height * pixelRatio);
+    CAMetalLayer *layer = (CAMetalLayer *)self.view.layer;
+    CGSize tsize             = CGSizeMake(static_cast<int>(size.width * pixelRatio),
+                                         static_cast<int>(size.height * pixelRatio));
+    //self.view.contentScaleFactor = pixelRatio;
```
Comment (minggo) [inline templates/ios/ViewController.mm:64]: 没用的代码就删除吧

File: templates/ios/ViewController.mm [hunk 2]
```diff
@@ -51,31 +48,21 @@ - (BOOL)prefersHomeIndicatorAutoHidden {
...
-            orientation = cc::Device::Orientation::LANDSCAPE_RIGHT;
-            break;
-        default:
-            break;
-    }
-    if (_lastOrientation != orientation) {
-        cc::EventDispatcher::dispatchOrientationChangeEvent((int) orientation);
-        _lastOrientation = orientation;
-    }
+    cc::EventDispatcher::dispatchOrientationChangeEvent((int) orientation);
```
Comment (minggo) [inline templates/ios/ViewController.mm:56]: 也要 static_cast 吧

**Comment by timlyeee** [comment]
restart ci

**Comment by timlyeee** [comment]
Update 2022.03.08:

This Pull request didn't provide enough information to developers.

Without the fix, when the developer builds an app with several orientations, the display effect for the start orientation is good, but once the orientation is changed, the screen is stretched and touchpoints are lost. 
image
image

**Pull request merged**